source/src_lcao/wfc_dm_2d.cpp

```diff
@@ -28,117 +30,190 @@ void Wfc_Dm_2d::init()
 	}
 }
 
-void Wfc_Dm_2d::cal_dm(const ModuleBase::matrix &wg)
+
+//for gamma_only
+void Wfc_Dm_2d::cal_dm(const ModuleBase::matrix& wg,
+    std::vector<ModuleBase::matrix>& wfc_gamma,
+    std::vector<ModuleBase::matrix>& dm_gamma)
 {
 	ModuleBase::TITLE("Wfc_Dm_2d", "cal_dm");
 	
 	#ifdef TEST_DIAG
-	{
-		static int istep=0;
-		std::ofstream ofs("wfc_"+ModuleBase::GlobalFunc::TO_STRING(istep++)+"_"+ModuleBase::GlobalFunc::TO_STRING(GlobalV::MY_RANK));
-		if(GlobalV::GAMMA_ONLY_LOCAL)
-		{
-			ofs<<wfc_gamma<<std::endl;
-		}
-		else
-		{
-			ofs<<wfc_k<<std::endl;
-		}
-	}
+    static int istep=0;
```

Comment:
In this file, some code has been repeated at lease twice, maybe developer should use inline function for that, and annotation is important either.